src/content/docs/en/reference/dev-overlay-plugin-reference.mdx

```diff
@@ -247,26 +247,87 @@ The dev overlay includes a set of web components that can be used to build plugi
 
 ### `astro-dev-overlay-window`
 
-Shows a window with optionally a title and an icon.
-
-`window-title` is a string that will be shown at the top of the overlay window. `window-icon` can either be a string of a SVG file or an icon from [the icon list](#icons).
+Shows a window.
 
 The slot of the component will be used as the content of the window.
 
 ```html
-<astro-dev-overlay-window window-title="MyWindows" window-icon="astro:logo">
+<astro-dev-overlay-window>
   <p>My content</p>
 </astro-dev-overlay-window>
 ```
 
+When building a window using JavaScript, remember that slotted content must go inside the light DOM of the component.
+
+```js
+const myWindow = document.createElement('astro-dev-overlay-window');
+const myContent = document.createElement('p');
+myContent.textContent = 'My content';
+
+// GOOD:
+myWindow.appendChild(myContent);
+
+// BAD:
+myWindow.shadowRoot.appendChild(myContent);
+```
+
+### `astro-dev-overlay-button`
+
+Shows a button.
+
+The slot of the component will be used as the content of the button.
+
+```js
+const myButton = document.createElement('astro-dev-overlay-button');
+myButton.textContent = 'Click me!';
+myButton.buttonStyle = "purple";
+myButton.size = "medium";
+
+myButton.addEventListener('click', () => {
+  console.log('Clicked!');
+});
+```
+
+#### `size`
+
+The size of the button, possible values are `small`, `medium` and `large`.
+
+#### `button-style`
+
+The style of the button, possible values are `ghost`, `outline`, `purple`, `gray`, and `red`. When using `ghost`, only the content of the button will be shown.
+
+In JavaScript, this property can be set using the `buttonStyle` property, as to avoid conflict with the native `style` property.
+
+### `astro-dev-overlay-badge`
+
+Shows a badge.
+
+The slot of the component will be used as the content of the badge.
+
+```html
+<astro-dev-overlay-badge>My badge</astro-dev-overlay-badge>
+```
+
+#### `size`
+
+The size of the badge, possible values are `small` and `large`.
+
+#### `badge-style`
+
+The style of the badge, possible values are `purple`, `gray`, `red`, `green`, and `yellow`.
```

Comment:
```suggestion
The style (color) of the badge (`purple`, `gray`, `red`, `green`, `yellow`).
```
It feels a tiny bit weird to have the property be named `style` when the options are all colours. I might therefore add (color) in brackets as a bit of extra context.

Reply:
Yeah, I chose `style` for the convention and in case in the future there's actual styles in there. It is wonky!